azurerm/azurerm_management_group/child/main.tf

```diff
@@ -0,0 +1,37 @@
+# Summary: A Management Group as a child of another Management Group
+
+# Documentation: https://www.terraform.io/docs/language/settings/index.html
+terraform {
+  required_version = ">= 0.14.0"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 2.0"
+    }
+  }
+}
+
+# Documentation: https://www.terraform.io/docs/language/values/variables.html
+variable "azure_subscription_id" {
+  type = string
+}
+
+# Documentation: https://www.terraform.io/docs/language/values/variables.html
+variable "changeme_parent_management_group_id" {
+  type = string
+}
```

Comment:
When using a variable without a default value it will require the user to input this value when running `./run.sh`.
It also breaks the CI "Run Azure Examples", see: https://github.com/ContainerSolutions/terraform-examples/runs/3073983061
```
Run /home/runner/work/terraform-examples/terraform-examples/bin/run_azurerm_examples.sh
azurerm/azurerm_management_group/child
./run.sh

var.changeme_parent_management_group_id
╷
│ Error: No value for required variable
│ 
│   on main.tf line 20:
│   20: variable "changeme_parent_management_group_id" ***
│ 
│ The root module input variable "changeme_parent_management_group_id" is not
│ set, and has no default value. Use a -var or -var-file command line
│ argument to provide a value for this variable.
```

We need to make this example work without user input.

----

I'm not sure about the intent here and how exactly it works in Azure. But maybe creating both `child_management_group` and `child_management_group` here (in the same `main.tf`) would help?

Reply:
TBH it would need to be provided as the MG depends on this value and there is no universal default solution for this.

We have to do sg with this if it breaks CI though. What if I disabled CI for this use case, and we find out how to handle examples when they have dependencies?

Reply:
Could we do something like:

```
resource "azurerm_management_group" "changeme_parent_management_group" {
  display_name = "changeme-parent-management-group-display-name"
}

resource "azurerm_management_group" "changeme_child_management_group" {
  display_name               = "changeme-child-management-group-display-name"
  parent_management_group_id = azurerm_management_group.changeme_parent_management_group.id
}
```

So, to create both resources here, instead of having dependencies between the examples. 
What do you think?

Reply:
will do definitely, I want to rebase each branch as they are merged, so this will wait until we merge the rest